CLN: Remove unnecessary sys.version_info checks

gfyoung opened the pull request:
Title is self-explanatory.

gfyoung committed:
CLN: Remove unnecessary sys.version_info checks

jbrockmendel commented:
I think versioned is vendored

gfyoung commented:
@jbrockmendel : Given the [history of changes](https://github.com/pandas-dev/pandas/commits/master/versioneer.py) we have made to it, I think it's fine to modify.

WillAyd commented on versioneer.py:
```diff
@@ -8,7 +8,7 @@
 * Brian Warner
 * License: Public Domain
-* Compatible With: python2.6, 2.7, 3.2, 3.3, 3.4, and pypy
+* Compatible With: python3.5+, and pypy
```
Actually I suppose this should read Python 3.6+, but given this is part of the licensing (?) probably best to just leave it alone

gfyoung replied:
This is just documentation, not licensing, and we've made enough changes to this files as is.

Now that I look at this, I actually will just drop this line because I think we can safely assume that the file is going to be compatible with whatever versions of Python `pandas` itself supports.

gfyoung committed:
CLN: Drop compatiblity line from versioneer

We can safely assume this file is going to
be (or should be) compatible with whatever
Python versions pandas itself supports.

simonjayhawkins commented on versioneer.py:
```diff
@@ -464,9 +463,10 @@ def run_command(commands, args, cwd=None, verbose=False, hide_stderr=False):
     stdout = p.communicate()[0].strip()
-    if sys.version_info[0] >= 3:
-        stdout = stdout.decode()
+    stdout = stdout.decode()
```
`stdout = p.communicate()[0].strip().decode()` is probably better to avoid mypy assigment of string to bytes errors if types are added.

gfyoung replied:
A bit out of scope for this PR, but sure, why not.

simonjayhawkins commented on versioneer.py:
```diff
@@ -561,9 +561,10 @@ def run_command(commands, args, cwd=None, verbose=False, hide_stderr=False):
     stdout = p.communicate()[0].strip()
-    if sys.version_info[0] >= 3:
-        stdout = stdout.decode()
+    stdout = stdout.decode()
```
same

gfyoung replied:
Same reply as [above](https://github.com/pandas-dev/pandas/pull/29210#discussion_r338909534)

gfyoung committed:
CLN: Mollify the mypy for typing

jreback commented:
thanks @gfyoung
